### DISCUSSION TITLE
Question about owstaxId<PropertyName> managed property

### DISCUSSION THREAD

**Discussion by bckrma** [microsoft-search/pnp-modern-search#2979] [Q&A] [answered]
Hello Community,
I have a problem with the "owstaxId<PropertyName>" managed property for a managed metadata column, which is empty for a lot of items in my lists and libraries. I am not sure if this is a problem with my configuration, or something i need to contact the microsoft support for. So sorry in advance if this is the wrong place for this question.

On a few sites, i have a site level term store configured, with one term set and a lot of child terms. This term set is used in a managed metadata column called "ManagedData".

My goal is to use the search verticle and search result webparts to only display items, where the "ManagedData" column is a child of a specific term.
To do this, i use this query in my search result webpart:
`{searchTerms} ParentLink:{/LibraryName}
owstaxIdManagedData:{verticals.value}`

My pnp search verticle webpart is configured like this:
|Tab Name| Tab Value|
|---|---|
|All| GTSet\|#bbbe4847-df61-4a5f-8c99-259421f87878
|Filtered|GPP\|#37ee358d-0cd7-49df-8ca2-b139911078e7

As far as i understand the "owstaxIdManagedData" column was created automatically and should contain the "path" to the term using the unique identifier for each label. For example:
`GP0|#45c76bf9-8e2a-4f83-9f32-b5g69f8218b0;L0|#045c76bf9-8e2a-4f83-8t96-b9a64f8218b0|<Term Label>;GTSet|#bbbe4847-df61-4a5f-8c99-259421f77777;GPP|#8ba73728-d19d-4a16-89865-7309500719da`

My search query using the value from the verticle webparts works for some items, but the results are incomplete.
I have used the sharepoint search query tool with a search query using the term label and i have noticed that the "owstaxIdManagedData" property is empty for a lot of items. Although these items have values for the "ManagedData" column and i was able to query for them using the term label.

I have already re-indexed the sites, lists and libraries and waited for multiple days. But the property is still empty.

Is this a problem with the sharepoint search, or is something wrong with my approach to use the "owstaxIdManagedData" property for this?
Thanks a lot!

**Comment by wobba** [comment]
Is the the column in your list/library a site column? That should ensure you get an automatically managed property matching the column. And you can try to re-index the list/library and see if the values are populated. Or create a new item.

**Reply by bckrma** [reply]
Yes it is a site column and the _owstaxIdManagedData_ managed property was also created.
One thing i forgot to mention is that the list and library items were migrated into the site using sharegate, if this has any influence.
I have also tried to re-index the list and libraries (_List Settings - Advanced Settings - Reindex list_) and the site (_Site Settings - Search and Offline Availability - Reindex site_) but even after a few days the _owstaxIdManagedData_ wasnt populated.

**Reply by wobba** [reply, marked as answer]
Data being migrated (or created via API's) are indexed with less priority.. give it some days and recheck. And don't keep hitting the reindex button - as that will have an adverse effect on re-queuing content.

**Reply by bckrma** [reply]
Ah thats good to know, i will give it a few more days then. 
Thanks a lot, specially for all the work you put into this project!

**Reply by sog-jfiske** [reply]
Hi, just checking in on this. Did it sort itself out after waiting a few days?

I'm facing a similar problem with some of my documents. Most times everything works fine, but now and then there's a document where some of the owstaxId[columnname] props never gets a value (waiting several weeks without success). I'm yet to find a reason for this since all documents have exactly the same properties and are created using the same content type.

The only non-standard procedure in the chain is that I'm copying the document from one site to another using the copy action in Power Automate, which probably is using API's in the back end to create the copied document at the destination site. The destination site has the same content type (provisioned through content type hub) and when I check some of my manually created managed props (mapped to ows_[columnname]) they have the right value. It's just the auto created owstaxId[columnname] that lacks value sometimes.
